src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -1407,7 +1410,7 @@ internal bool IsNonAttributedType
                     _isKeyValuePairAdapter = true;
                     _keyValuePairGenericArguments = type.GetGenericArguments();
                     _keyValuePairCtorInfo = type.GetConstructor(Globals.ScanAllMembers, new Type[] { Globals.TypeOfKeyValuePair.MakeGenericType(_keyValuePairGenericArguments) });
```

Comment:
What happens if you also use `GetMemberWithSameMetadataDefinitionAs` for the constructor? Does the linker still need the DynamicDependency above?

Reply:
Yeah that does seem to make it work. I'll switch to use that instead of Dynamic Dependency.